python/GafferSceneUI/SetEditor.py

```diff
@@ -0,0 +1,379 @@
+
+		if self.__plug is None :
+			return
+
+		selection = self.__selectedSetNames()
+		if len( selection ) > 0 :
+			data = IECore.StringVectorData( selection )
+			self.__plug.ancestor( Gaffer.ApplicationRoot ).setClipboardContents( data )
+
+	def __getSetMembers( self, *unused ) :
+
+		if self.__plug is None :
+			return
+
+		result = IECore.PathMatcher()
+		with Gaffer.Context( self.getContext() ) :
+			for setName in self.__selectedSetNames() :
+				result.addPaths( self.__plug.set( setName ).value )
+
+		return result
+
+	def __selectSetMembers( self, *unused ) :
+
+		locations = self.__getSetMembers()
+		if not locations.isEmpty() :
+			ContextAlgo.setSelectedPaths( self.getContext(), locations )
+
+	def __copySetMembers( self, *unused ) :
+
+		if self.__plug is None :
+			return
+
+		locations = self.__getSetMembers()
+		if not locations.isEmpty() :
+			data = IECore.StringVectorData( locations.paths() )
+			self.__plug.ancestor( Gaffer.ApplicationRoot ).setClipboardContents( data )
+
+GafferUI.Editor.registerType( "SetEditor", SetEditor )
+
+##########################################################################
+# _SearchFilterWidget
+##########################################################################
+
+class _SearchFilterWidget( GafferUI.PathFilterWidget ) :
+
+	def __init__( self, pathFilter ) :
+
+		self.__patternWidget = GafferUI.TextWidget()
+		GafferUI.PathFilterWidget.__init__( self, self.__patternWidget, pathFilter )
+
+		self.__patternWidget._qtWidget().setPlaceholderText( "Filter..." )
+
+		self.__patternWidget.editingFinishedSignal().connect( Gaffer.WeakMethod( self.__patternEditingFinished ), scoped = False )
+		self.__patternWidget.dragEnterSignal().connectFront( Gaffer.WeakMethod( self.__dragEnter ), scoped = False )
+		self.__patternWidget.dragLeaveSignal().connectFront( Gaffer.WeakMethod( self.__dragLeave ), scoped = False )
+		self.__patternWidget.dropSignal().connectFront( Gaffer.WeakMethod( self.__drop ), scoped = False )
+
+		self.__context = None
+		self.__scene = None
+
+		self._updateFromPathFilter()
+
+	def setContext( self, context ) :
+
+		self.__context = context
+
+	def setScene( self, scene ) :
+
+		self.__scene = scene
+
+	def _updateFromPathFilter( self ) :
+
+		self.__patternWidget.setText( self.pathFilter().getMatchPattern() )
+
+	def __patternEditingFinished( self, widget ) :
+
+		self.pathFilter().setMatchPattern( self.__patternWidget.getText() )
+
+	def __dragEnter( self, widget, event ) :
+
+		self.__patternWidget.setHighlighted( True )
+
+		return True
+
+	def __dragLeave( self, widget, event ) :
+
+		self.__patternWidget.setHighlighted( False )
+
+		return True
+
+	def __drop( self, widget, event ) :
+
+		if isinstance( event.data, IECore.StringVectorData ) and len( event.data ) > 0 :
+			if event.data[0].startswith( "/" ) :
+				# treat as paths, return all sets those paths are members of
+				setNames = self.__getSetNamesFromPaths( event.data )
```

Comment:
I'm a bit wary of how open-ended the cost of calling this might be, given that we're blocking the UI till it's done. I guess it is a very specific action to perform, and unlikely to be done accidentally though. Any merit in using a BackgroundTaskDialogue so that it can be cancelled if necessary?

Reply:
Great idea! I've added the BackgroundTaskDialogue in 4f0cbf0.